src/io.jl

```diff
@@ -80,44 +80,49 @@ end
 
 function _load(png_ptr, info_ptr; gamma::Union{Nothing,Float64}=nothing, expand_paletted::Bool=false)
     png_read_info(png_ptr, info_ptr)
+    flags = png_get_valid(
+        png_ptr, info_ptr,
+        PNG_INFO_sRGB | PNG_INFO_gAMA | PNG_INFO_tRNS | PNG_INFO_cHRM | PNG_INFO_bKGD | PNG_INFO_PLTE
+    )
+    valid_sRGB = (flags & PNG_INFO_sRGB) != 0
+    valid_gAMA = (flags & PNG_INFO_gAMA) != 0
+    valid_tRNS = (flags & PNG_INFO_tRNS) != 0
+    valid_cHRM = (flags & PNG_INFO_cHRM) != 0
+    valid_bKGD = (flags & PNG_INFO_bKGD) != 0
+    valid_PLTE = (flags & PNG_INFO_PLTE) != 0
 
     width = png_get_image_width(png_ptr, info_ptr)
     height = png_get_image_height(png_ptr, info_ptr)
-    color_type_orig = png_get_color_type(png_ptr, info_ptr)
-    color_type = color_type_orig
-    bit_depth_orig = png_get_bit_depth(png_ptr, info_ptr)
-    bit_depth = bit_depth_orig
+    color_type = color_type_orig = png_get_color_type(png_ptr, info_ptr)
+    bit_depth = bit_depth_orig = png_get_bit_depth(png_ptr, info_ptr)
     num_channels = png_get_channels(png_ptr, info_ptr)
     interlace_type = png_get_interlace_type(png_ptr, info_ptr)
 
-    # TODO: verify this is needed
-    backgroundp = png_color_16p()
-    if png_get_bKGD(png_ptr, info_ptr, Ptr{png_color_16p}(backgroundp)) != 0
-        png_set_background(png_ptr, backgroundp, PNG_BACKGROUND_GAMMA_FILE, 1, 1.0)
+    if valid_bKGD
+        backgroundp = png_color_16p()
+        if png_get_bKGD(png_ptr, info_ptr, Ptr{png_color_16p}(backgroundp)) != 0
+            png_set_background(png_ptr, backgroundp, PNG_BACKGROUND_GAMMA_FILE, 1, 1.)
```

Comment:
Biggest possible nit: Can this be `1.0` instead of `1.`?

Reply:
Ha sorry, my hate for trailing or leading zeros has landed in this PR ;). Restored.